### PR TITLE
OSDOCS-8026: Network Observability Z-Stream Release Notes - 1.4.1

### PR DESCRIPTION
**Pull request by skrthomas**
Version(s):

4.11+
Issue:

https://issues.redhat.com/browse/OSDOCS-8206
Link to docs preview:

https://66302--docspreview.netlify.app/openshift-enterprise/latest/networking/network_observability/network-observability-operator-release-notes
QE review:
- [X] QE has approved this change.

Additional information:

### CODE REVIEW

File: networking/network_observability/network-observability-operator-release-notes.adoc [hunk 1]
```diff
@@ -12,6 +12,22 @@ The Network Observability Operator enables administrators to observe and analyze
...
+
+* link:https://access.redhat.com/errata/RHSA-2023:122114[2023:122114 Network Observability Operator 1.4.1]
+
+=== CVEs
+* link:https://access.redhat.com/security/cve/cve-2023-44487[2023-44487]
+* link:https://access.redhat.com/security/cve/cve-2023-39325[2023-39325]
+
+== Bug fix
+
+* Previously, when sending to Kafka, the Kafka message key was ignored causing an error with connection tracking. Now the key is used for partitioning, so each flow from the same connection are sent to the same processor. (link:https://issues.redhat.com/browse/NETOBSERV-926[*NETOBSERV-926*])
```
Comment (jotak) [inline networking/network_observability/network-observability-operator-release-notes.adoc:27]: just wondering: should we say this was previously mentioned as a known issue, hence that known issue can be ignored now? Or we can consider this is implicit.

File: networking/network_observability/network-observability-operator-release-notes.adoc [hunk 2]
```diff
@@ -12,6 +12,22 @@ The Network Observability Operator enables administrators to observe and analyze
...
 For an overview of the Network Observability Operator, see xref:../../networking/network_observability/network-observability-overview.adoc#dependency-network-observability[About Network Observability Operator].
+[id="network-observability-operator-release-notes-1-4-1"]
+== Network Observability Operator 1.4.1
+The following advisory is available for the Network Observability Operator 1.4.1:
+
+* link:https://access.redhat.com/errata/RHSA-2023:122114[2023:122114 Network Observability Operator 1.4.1]
+
+=== CVEs
+* link:https://access.redhat.com/security/cve/cve-2023-44487[2023-44487]
+* link:https://access.redhat.com/security/cve/cve-2023-39325[2023-39325]
```
Comment (jotak) [inline networking/network_observability/network-observability-operator-release-notes.adoc:23]: I've added more CVE fixes to the list, this morning:
- [CVE-2023-29406](https://access.redhat.com/security/cve/CVE-2023-29406)
- [CVE-2023-29409](https://access.redhat.com/security/cve/CVE-2023-29409)
- [CVE-2023-39318](https://access.redhat.com/security/cve/CVE-2023-39318)
- [CVE-2023-39319](https://access.redhat.com/security/cve/CVE-2023-39319)
- [CVE-2023-39322](https://access.redhat.com/security/cve/CVE-2023-39322)

**Review by jotak — APPROVED** [review]
lgtm!

File: networking/network_observability/network-observability-operator-release-notes.adoc (hunk 1 above)
Comment (skrthomas) [reply networking/network_observability/network-observability-operator-release-notes.adoc:33]: I actually think the known issue should be left in 1.4 and then mention that it was previously a known issue in 1.4.1 bug fix note, like your first question was asking. I don't think its right to edit out the known issue.

Comment (skrthomas) [reply networking/network_observability/network-observability-operator-release-notes.adoc]: Just updated this!

File: networking/network_observability/network-observability-operator-release-notes.adoc [hunk 3]
```diff
@@ -12,6 +12,28 @@ The Network Observability Operator enables administrators to observe and analyze
...
+* link:https://access.redhat.com/security/cve/cve-2023-29406[2023-29406]
+* link:https://access.redhat.com/security/cve/CVE-2023-29409[2023-29409]
+* link:https://access.redhat.com/security/cve/cve-2023-39322[2023-39322]
+* link:https://access.redhat.com/security/cve/cve-2023-39318[2023-39318]
+* link:https://access.redhat.com/security/cve/cve-2023-39319[2023-39319]
+* link:https://access.redhat.com/security/cve/cve-2023-39321[2023-39321]
+
+== Bug fixes
+
+* In 1.4, there was a known issue when sending network flow data to Kafka. The Kafka message key was ignored, causing an error with connection tracking. Now the key is used for partitioning, so each flow from the same connection are sent to the same processor. (link:https://issues.redhat.com/browse/NETOBSERV-926[*NETOBSERV-926*])
```
Comment (mburke5678) [inline networking/network_observability/network-observability-operator-release-notes.adoc:33]: each.. is sent
```suggestion
* In 1.4, there was a known issue when sending network flow data to Kafka. The Kafka message key was ignored, causing an error with connection tracking. Now the key is used for partitioning, so each flow from the same connection is sent to the same processor. (link:https://issues.redhat.com/browse/NETOBSERV-926[*NETOBSERV-926*])
```

File: networking/network_observability/network-observability-operator-release-notes.adoc [hunk 4]
```diff
@@ -12,6 +12,28 @@ The Network Observability Operator enables administrators to observe and analyze
...
+
+* In 1.4, the "Inner" flow direction was introduced to account for flows between pods running on the same node. Flows with the "Inner" direction were not taken into account in the generated Prometheus metrics derived from flows, resulting in under-evaluated bytes and packets rates.
+Now, derived metrics are including flows with the "Inner" direction, thus providing correct bytes and packets rates. (link:https://issues.redhat.com/browse/NETOBSERV-1344[*NETOBSERV-1344*])
```
Comment (mburke5678) [inline networking/network_observability/network-observability-operator-release-notes.adoc:36]: Minimalism
```suggestion
Now, derived metrics are including flows with the "Inner" direction, providing correct bytes and packets rates. (link:https://issues.redhat.com/browse/NETOBSERV-1344[*NETOBSERV-1344*])
```

File: networking/network_observability/network-observability-operator-release-notes.adoc [hunk 5]
```diff
@@ -73,7 +95,7 @@ You can now export eBPF-enriched network flows to the IPFIX collector. For more
...
+In the 1.4 release of the Network Observability Operator, eBPF tracepoint hooks are used to enable packet drop tracking. You can now detect and analyze the cause for packet drops and make decisions to optimize network performance. In 4.14+, both host drops and OVS drops are detected. In 4.13, only host drops are detected. For more information, see xref:../../networking/network_observability/observing-network-traffic#network-observability-pktdrop-overview_nw-observe-network-traffic[Configuring packet drop tracking] and xref:../../networking/network_observability/observing-network-traffic#network-observability-packet-drops_nw-observe-network-traffic[Working with packet drops].
```
Comment (mburke5678) [inline networking/network_observability/network-observability-operator-release-notes.adoc:98]: To be clear on what product we are talking about
```suggestion
In the 1.4 release of the Network Observability Operator, eBPF tracepoint hooks are used to enable packet drop tracking. You can now detect and analyze the cause for packet drops and make decisions to optimize network performance. In {product-title} 4.14 and later, both host drops and OVS drops are detected. In {product-title} 4.13, only host drops are detected. For more information, see xref:../../networking/network_observability/observing-network-traffic#network-observability-pktdrop-overview_nw-observe-network-traffic[Configuring packet drop tracking] and xref:../../networking/network_observability/observing-network-traffic#network-observability-packet-drops_nw-observe-network-traffic[Working with packet drops].
```

**Comment by mburke5678** [comment]
@skrthomas A few nits. Otherwise LGTM

File: networking/network_observability/network-observability-operator-release-notes.adoc (hunk 5 above)
Comment (skrthomas) [reply networking/network_observability/network-observability-operator-release-notes.adoc:98]: These changes are actually part of https://github.com/openshift/openshift-docs/pull/66047...they won't be relevant until 4.14 so I'm going to redact them here. Not sure actually how they ended up here

File: networking/network_observability/network-observability-operator-release-notes.adoc [hunk 6]
```diff
@@ -12,6 +12,28 @@ The Network Observability Operator enables administrators to observe and analyze
...
+* link:https://access.redhat.com/security/cve/cve-2023-39322[2023-39322]
+* link:https://access.redhat.com/security/cve/cve-2023-39318[2023-39318]
+* link:https://access.redhat.com/security/cve/cve-2023-39319[2023-39319]
+* link:https://access.redhat.com/security/cve/cve-2023-39321[2023-39321]
+
+== Bug fixes
+
+* In 1.4, there was a known issue when sending network flow data to Kafka. The Kafka message key was ignored, causing an error with connection tracking. Now the key is used for partitioning, so each flow from the same connection are sent to the same processor. (link:https://issues.redhat.com/browse/NETOBSERV-926[*NETOBSERV-926*])
+
+* In 1.4, the "Inner" flow direction was introduced to account for flows between pods running on the same node. Flows with the "Inner" direction were not taken into account in the generated Prometheus metrics derived from flows, resulting in under-evaluated bytes and packets rates.
```
[the remark this answers is not in the document]
Comment (skrthomas) [reply networking/network_observability/network-observability-operator-release-notes.adoc:35]: It is JSON, so I think when we refer to it, it should have markup. Its also a UI Item in the Traffic Flows view, mentioned here under Advanced Filters, although now I'm seeing this markup wasn't properly applied and will need to be fixed:https://docs.openshift.com/container-platform/4.13/networking/network_observability/observing-network-traffic.html#network-observability-quickfilternw-observe-network-traffic

Maybe that reference should markup the JSON too...I can open a docs bug to investigate and at least fix that bad markup.

**Pull request merged**